### PR TITLE
DB2 updates

### PR DESCRIPTION
**Pull request by dwadler**
Fix DB2 SQL generation to use units like METER for DWITHIN and BEYOND. Previously, units were ignored and distance was treated as being in the coordinate system units.

Updated online testcase to correctly test DWITHIN and BEYOND.

### CODE REVIEW

File: modules/plugin/jdbc/jdbc-db2/src/main/java/org/geotools/data/db2/DB2FilterToSQL.java [hunk 1]
```diff
@@ -130,6 +130,29 @@
...
     private static WKTWriter wktWriter = new WKTWriter();
 
+    /**
+     * Conversion factor from common units to meter
+     */
+    private static final Map<String, Double> UNITS_MAP = new HashMap<String, Double>() {
+        {
+            put("kilometers", 1000.0);
+            put("kilometer", 1000.0);
+			put("meters", 1.0);
```
Comment (aaime) [inline modules/plugin/jdbc/jdbc-db2/src/main/java/org/geotools/data/db2/DB2FilterToSQL.java]: This line uses tabs, please switch to spaces (yes, there is old code that's not properly formatted, for newer code please follow the coding conventions, and no, do not mass reformat files)

**Comment by aaime** [comment]
Thank you. I don't have experience with DB2 so cannot review the changes or test them, @mcrmcr do you have time to have a look?

**Comment by dwadler** [comment]
Who is @mcrmcr? I did modify the existing online test for DWITHIN to perform a useful test and added a test for BEYOND which ran successfully. There are some other tests that are failing but they were failing before my change - I'm looking into why.

**Comment by aaime** [comment]
@mcrmcr  is the current maintainer of jdbc-db2 (check the module history, https://github.com/geotools/geotools/commits/master/modules/plugin/jdbc/jdbc-db2/src/main )

**Comment by dwadler** [comment]
Does @mcrmcr have a name? I thought Christian Mueller was maintaining the DB2 support. I was the original author in 2005.

**Comment by aaime** [comment]
@mcrmcr is Christian Mueller indeed, the current maintainer. He's the one that has to be comfortable with the changes. Unless he steps down and you become the new maintainer.

**Comment by mcrmcr** [comment]
Merged

**Pull request merged**